packages/flame/lib/src/components/sprite_component.dart

```diff
@@ -15,12 +15,46 @@ export '../sprite.dart';
 class SpriteComponent extends PositionComponent
     with HasPaint
     implements SizeProvider {
+  /// When set to true, the component is auto-resized to match the
+  /// size of underlying sprite.
+  late bool _autoResize;
+
+  /// Returns current value of auto resize flag.
+  bool get autoResize => _autoResize;
+
+  /// Sets the given value of autoResize flag. Will update the [size]
+  /// to fit srcSize of [sprite] if set to  true.
+  set autoResize(bool value) {
+    _autoResize = value;
+    if (value) {
+      size.setFrom(sprite?.srcSize ?? Vector2.zero());
```

Comment:
It should probably do a null check outside here instead and use `setZero` if it's null so that no new vector is created

Reply:
Ah, just like you do further down

Reply:
Done.